### DISCUSSION TITLE
wasi-motor vs wasmer-js

### DISCUSSION THREAD

**Discussion by sergree** [taybenlor/runno#218] [Q&A] [answered]
Hey, @taybenlor!

Thank you very much for this project! I'm really inspired by the capabilities of WASI.
You've done a wonderful job.

I am currently exploring possible ways to launch WASI applications inside the browser. Besides your wasi-motor, I found another similar project: [wasmerio/wasmer-js](https://github.com/wasmerio/wasmer-js). But it looks like this project is older and hasn't been updated for a while.

I would like to hear your professional opinion, what is the difference between their project and yours and what should be used for development now? 

I'm sorry if my question seemed silly, I have almost no experience in WASI. And there is very little information about this topic.

Thank you in advance!

**Comment by taybenlor** [comment]
Hey @sergree! Thanks for the nice words.

`@runno/wasi-motor` is definitely not production ready yet. I've got to put some work into making it usable outside of the runno.dev website.

The `wasmerio/wasmer-js` package I believe is their older deprecated library for executing WASI binaries in the browser. I'd recommend you have a look at `@wasmer/wasi` (https://www.npmjs.com/package/@wasmer/wasi).

I'd really like to get my `wasi-motor` project into a state where it can be used by others, but unfortunately right now it's more of a toy.

Happy to answer any questions you have about WASI and the ecosystem either here or by email!

Cheers,
Ben

**Reply by sergree** [reply]
Thank you for your reply! It is very helpful.
I hope that the WASI ecosystem will actively evolve. It can be a real game changer.

🙏

**Comment by taybenlor** [comment, marked as answer]
I'd like to supercede this answer with an update!

[`@runno/wasi`](https://www.npmjs.com/package/@runno/wasi) is now ready for public use. You can see details on how to use it [on NPM](https://www.npmjs.com/package/@runno/wasi) or here [on github](https://github.com/taybenlor/runno/blob/main/packages/wasi/README.md).

My goal with `@runno/wasi`, and runno in general, is to build WASI tools for the web. `@runno/runtime` is now shipping a `<runno-wasi>` web component that you can use directly. I'll be updating the website soon with more details!
